infra/order-generator/index.ts

```diff
@@ -0,0 +1,239 @@
+    secretId: pinataJwtSecret.id,
+    secretString: pinataJWT,
+  });
+
+  const rpcUrlSecret = new aws.secretsmanager.Secret(`${serviceName}-rpc-url`);
+  new aws.secretsmanager.SecretVersion(`${serviceName}-rpc-url`, {
+    secretId: rpcUrlSecret.id,
+    secretString: ethRpcUrl,
+  });
+
+  const repo = new awsx.ecr.Repository(`${serviceName}-repo`, {
+    forceDelete: true,
+    lifecyclePolicy: {
+      rules: [
+        {
+          description: 'Delete untagged images after N days',
+          tagStatus: 'untagged',
+          maximumAgeLimit: 7,
+        },
+      ],
+    },
+  });
+
+  const authToken = aws.ecr.getAuthorizationTokenOutput({
+    registryId: repo.repository.registryId,
+  });
+
+  let buildSecrets = {};
+  if (githubTokenSecret !== undefined) {
+    buildSecrets = {
+      ...buildSecrets,
+      githubTokenSecret
+    }
+  }
+
+  const dockerTagPath = pulumi.interpolate`${repo.repository.repositoryUrl}:${dockerTag}`;
+
+  const image = new docker_build.Image(`${serviceName}-image`, {
+    tags: [dockerTagPath],
+    context: {
+      location: dockerDir,
+    },
+    platforms: ['linux/amd64'],
+    push: true,
+    dockerfile: {
+      location: `${dockerDir}/dockerfiles/order_generator.dockerfile`,
+    },
+    secrets: buildSecrets,
+    cacheFrom: [
+      {
+        registry: {
+          ref: pulumi.interpolate`${repo.repository.repositoryUrl}:cache`,
+        },
+      },
+    ],
+    cacheTo: [
+      {
+        registry: {
+          mode: docker_build.CacheMode.Max,
+          imageManifest: true,
+          ociMediaTypes: true,
+          ref: pulumi.interpolate`${repo.repository.repositoryUrl}:cache`,
+        },
+      },
+    ],
+    registries: [
+      {
+        address: repo.repository.repositoryUrl,
+        password: authToken.password,
+        username: authToken.userName,
+      },
+    ],
+  });
+
+  // const image = new awsx.ecr.Image(`${serviceName}-image`, {
+  //   repositoryUrl: repo.url,
+  //   context: dockerDir,
+  //   dockerfile: `${dockerDir}/dockerfiles/order_generator.dockerfile`,
+  //   imageTag: dockerTag,
+  //   platform: 'linux/amd64',
+  // });
+
+  // Security group allow outbound, deny inbound
+  const securityGroup = new aws.ec2.SecurityGroup(`${serviceName}-security-group`, {
+    name: serviceName,
+    vpcId,
+    egress: [
+      {
+        fromPort: 0,
+        toPort: 0,
+        protocol: '-1',
+        cidrBlocks: ['0.0.0.0/0'],
+        ipv6CidrBlocks: ['::/0'],
+      },
+    ],
+  });
+
+  // Create an execution role that has permissions to access the necessary secrets
+  const execRole = new aws.iam.Role(`${serviceName}-exec-1`, {
+    assumeRolePolicy: aws.iam.assumeRolePolicyForPrincipal({
+      Service: 'ecs-tasks.amazonaws.com',
+    }),
+    managedPolicyArns: [aws.iam.ManagedPolicy.AmazonECSTaskExecutionRolePolicy],
+  });
+
+  const execRolePolicy = new aws.iam.RolePolicy(`${serviceName}-exec-1`, {
```

Comment:
why add `1`? Is there another policy that collides?

Reply:
In the other repo I believe there was. Removed from here.